test/cpp/naming/gen_build_yaml.py

```diff
@@ -33,21 +39,107 @@ def _build_expected_addrs_cmd_arg(expected_addrs):
     out.append('%s,%s' % (addr['address'], str(addr['is_balancer'])))
   return ';'.join(out)
 
+def _data_for_type(r_type, r_data, common_zone_name):
+  if r_type in ['A', 'AAAA']:
+    return ' '.join(map(lambda x: '\"%s\"' % x, r_data))
+  if r_type == 'SRV':
+    assert len(r_data) == 1
+    target = r_data[0].split(' ')[3]
+    uploadable_target = '%s.%s' % (target, common_zone_name)
+    uploadable = r_data[0].split(' ')
+    uploadable[3] = uploadable_target
+    return '\"%s\"' % ' '.join(uploadable)
+  if r_type == 'TXT':
```

Comment:
It would be really useful to have a stand-alone script that would convert a service config string in JSON form into the form needed for a DNS zone file (which is presumably the same form used in gcloud).  This would be used by folks in the OSS world that want to encode the service config in their DNS server.  Can you refactor this to provide such a utility?

Feel free to do this in a separate PR.  It doesn't need to block this.

Reply:
SG, I think that gcloud has an alternate mode of uploading records that involves pointing it to a zone file, rather than specifying each record via command line parameters. I might try changing this around to use such a JSON -> TXT script a part of creating a zone file.